### DISCUSSION TITLE
Update Python to v3.12.2-amd64

### DISCUSSION THREAD

**Discussion by kstingel** [leokhoa/laragon#687] [General]
Problem: the only `.zip` archive available on Python website is the embedded version (which doesn't have the usual directory structure or included tools)  

Workaround: since I had pyenv-win on my system already,  
open a cmd prompt and type `pyenv update` to load recent python versions
run `pyenv install 3.12.2` - this installs the usual full python directory tree into the `{pyenv-win_path}/versions/3.12.2` folder
copy the new `3.12.2` from where pyenv-win installed it to `{laragon_root}/bin/python` folder
rename pasted folder to `python-3.12.2` and activate via taskbar icon

(untested) Configuration solution:
add `python-3.12.2=https://www.python.org/ftp/python/3.12.2/python-3.12.2-amd64.exe` to `usr/packages.conf` 
enable the new Python version via Tools > Quick Add...

**Comment by luisAntonioLAGS** [comment]
You can use the dot version of winpython, I update python with that package. For update move the folder into C:/laragon/bin/python
Url: https://winpython.github.io/
![imagen](https://github.com/leokhoa/laragon/assets/60680854/9b5b7de2-2fab-49e2-9527-f0114989ccbc)
![imagen](https://github.com/leokhoa/laragon/assets/60680854/f44222a6-f467-4c92-bf02-3f91705f0eb6)

**Reply by kstingel** [reply]
thanks for that info, will try that way next update
